### ISSUE TITLE
nginx.org debian repo has invalid PGP signature on InRelease

### ISSUE THREAD

**Issue by gStart9** [joinmarket-webui/jam-docker#127]
In [jam-startos](https://github.com/Start9Labs/jam-startos/blob/master/Dockerfile#L9C5-L9C62) we need wget, bash & tini so we run `apt update` which results in this error:

```
#0 11.23 Err:5 https://nginx.org/packages/mainline/debian bullseye InRelease
#0 11.23   The following signatures were invalid: EXPKEYSIG ABF5BD827BD9BF62 nginx signing key <signing-key@nginx.com>
#0 11.33 Reading package lists...
#0 14.05 W: GPG error: https://nginx.org/packages/mainline/debian bullseye InRelease: The following signatures were invalid: EXPKEYSIG ABF5BD827BD9BF62 nginx signing key <signing-key@nginx.com>
#0 14.05 E: The repository 'https://nginx.org/packages/mainline/debian bullseye InRelease' is not signed.
```

**Comment by gStart9** [comment]
From the [jam-docker Dockerfile](https://github.com/joinmarket-webui/jam-docker/blob/master/standalone/Dockerfile#L80-L82):
```
    && curl --silent https://nginx.org/keys/nginx_signing.key | \
    gpg --dearmor | tee /usr/share/keyrings/nginx-archive-keyring.gpg > /dev/null \
    && sh -c "echo 'deb [signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] https://nginx.org/packages/mainline/debian/ bullseye nginx' > /etc/apt/sources.list.d/nginx.list" \
```

Looks like they just updated their key in the last few months:

# sha256sum /usr/share/keyrings/nginx-archive-keyring.gpg
fc27fd284cceb4bf6c8ac2118dbb5e834590836f8d6ba3944da0e0451cbadeca  /usr/share/keyrings/nginx-archive-keyring.gpg
# sha256sum nginx_signing.key 
55385da31d198fa6a5012d40ae98ecb272a6c4e8fffffba94719ffd3e87de37a  nginx_signing.key

**Comment by remcoros** [comment]
the previous key expired June 14th, https://blog.nginx.org/blog/updating-pgp-key-for-nginx-software

**Comment by theborakompanioni** [comment]
Hey @gStart9 and @remcoros!

Thanks for reporting. I think I do not quite understand.. shouldn't the new signing key be automatically updated and imported? I would have thought no manual action is required here. 
Or is this specifically for older, already built images?

**Comment by remcoros** [comment]
It applies to all images that were built and published before that date.

When creating a custom Dockerfile that uses: `FROM ghcr.io/joinmarket-webui/jam-standalone:v0.2.0-clientserver-v0.9.11` which has the (old) key baked in, and then run `apt update` in the derived Dockerfile,

That's what https://github.com/Start9Labs/jam-startos/blob/master/Dockerfile does.

So either upstream will need to build and publish a new image, so downstream users have a new valid key. 
Or downstream users who use an image build before that date will need to redownload and apply the key in the derived Dockerfile.

@gStart9 to unblock and use the current jam image, you could copy those statements into the startos wrapper Dockerfile.

**Comment by theborakompanioni** [comment]
As a temporary workaround, new docker images have been pushed and are available here: [`v0.2.0-clientserver-v0.9.11-patch-20240622`](https://github.com/joinmarket-webui/jam-docker/pkgs/container/jam-dev-standalone/233829105?tag=v0.2.0-clientserver-v0.9.11-patch-20240622).

Please know that this is pushed in the `jam-dev-standalone` namespace, instead of `jam-standalone` .
```docker
FROM ghcr.io/joinmarket-webui/jam-dev-standalone:v0.2.0-clientserver-v0.9.11-patch-20240622
```

Hope this eases the development and removes the blocker @gStart9 @remcoros. Successfully tested on `regtest`.  Please let us know if everything works as expected for you.

On a new release, you can switch back to `jam-standalone` :raised_hands:
